### PR TITLE
#2618 Vaccines incoming sync [D2]

### LINKED ISSUE #2618: Vaccine module incoming sync logic
## Is your feature request related to a problem? Please describe.

Need incoming sync logic for all vaccine module new data types

## Describe the solution you'd like

- For incoming sync: https://github.com/openmsupply/mobile/blob/vaccine-development/src/sync/incomingSyncUtils.js

Incoming sync for
- `TemperatureLog`
- `TemperatureBreach`
- `MovementLog`
- `VaccineVialMonitorStatus`
- `VaccineVialMonitorStatusLog`
- `Location`
- `LocationType`
- `Sensor`
- Additional fields on: `TransactionBatch`, `StocktakeBatch` and `ItemBatch`, `Item`

## Describe alternatives you've considered

N/A

## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
#### NOTE: Branched from #1366 

Fixes #2618 

## Change summary

- Adds incoming sync logic for new vaccines schema

## Testing

N/A

### Related areas to think about

N/A

### CODE REVIEW

File: src/sync/incomingSyncUtils.js [hunk 1]
```diff
@@ -244,6 +262,9 @@ export const sanityCheckIncomingRecord = (recordType, record) => {
         'cost_price',
```
Comment (mark-prins) [inline src/sync/incomingSyncUtils.js]: 🤔 why does the `Store` have a cost and sell price?

File: src/sync/incomingSyncUtils.js [hunk 2]
```diff
@@ -421,6 +475,12 @@ export const createOrUpdateRecord = (database, settings, recordType, record) =>
         name: record.item_name,
         crossReferenceItem: database.getOrCreate('Item', record.cross_ref_item_ID),
         unit: database.getOrCreate('Unit', record.unit_ID),
+        doses: parseNumber(record.doses),
+        isVaccine: parseBoolean(record.boolean),
```
Comment (mark-prins) [inline src/sync/incomingSyncUtils.js]: Is `record.boolean` correct? I haven't checked the mSupply end, but surely some sore of vaccine-related property name here?

File: src/sync/incomingSyncUtils.js [hunk 3]
```diff
@@ -931,6 +1009,94 @@ export const createOrUpdateRecord = (database, settings, recordType, record) =>
...
+      });
+      break;
+    }
+    case 'VaccineVialMonitorStatus': {
+      database.update(recordType, {
+        id: record.ID,
+        description: record.description,
+        code: record.code,
+        level: parseNumber(record.level),
+        isActive: parseBoolean(record.boolean),
```
Comment (mark-prins) [inline src/sync/incomingSyncUtils.js]: there's `record.boolean` again. Is it ok?

Comment (joshxg) [reply src/sync/incomingSyncUtils.js]: spooky mistake! 285b03f58cda4382025894656533f442ab413c02

File: src/sync/incomingSyncUtils.js (hunk 2 above)
Comment (joshxg) [reply src/sync/incomingSyncUtils.js]: haha oops! nope, thanks! 285b03f58cda4382025894656533f442ab413c02

File: src/sync/incomingSyncUtils.js (hunk 1 above)
Comment (joshxg) [reply src/sync/incomingSyncUtils.js]: haha - I think it's just a weirdo diff

Comment (mark-prins) [reply src/sync/incomingSyncUtils.js]: that's good - I thought it was showing a huge gap in my understanding of the sync methods!

**Pull request merged**